CMakeLists.txt

```diff
@@ -46,7 +46,11 @@ endif()
 ##
 add_library(${NLOHMANN_JSON_TARGET_NAME} INTERFACE)
 add_library(${PROJECT_NAME}::${NLOHMANN_JSON_TARGET_NAME} ALIAS ${NLOHMANN_JSON_TARGET_NAME})
-target_compile_features(${NLOHMANN_JSON_TARGET_NAME} INTERFACE cxx_std_11)
+if (${CMAKE_VERSION} VERSION_LESS "3.8.0") 
+    target_compile_features(${NLOHMANN_JSON_TARGET_NAME} INTERFACE cxx_range_for)
+else()
+    target_compile_features(${NLOHMANN_JSON_TARGET_NAME} INTERFACE cxx_std_11)
+endif()
 
```

Comment:
Okay, I didn't know `cxx_std_11` forces specifically c++11 and not c++11 or higher. Lets try not to introduce different behaviour based on the cmake version.

Why did we settle on `cxx_range_for` to obtain the necessary c++11 features? The `auto` type is a c++11 feature, and it is available in cmake [version 3.1](https://cmake.org/cmake/help/v3.1/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#prop_gbl:CMAKE_CXX_KNOWN_FEATURES) as `cxx_auto_type`, wouldn't using that have the same effect as `cxx_range_for`, except that it supports cmake 3.1?

Reply:
`cxx_auto_type` and `cxx_range_for` are just two features which landed in C++11. Theoretically this library should specify all of them really used. In practice having one is enough as most features result in C++11 on the TOP-3 of the open-source compilers.

Reply:
But then, wouldn't the fix be to just use `cxx_auto_type` instead of `cxx_range_for` as that is supported by CMake 3.1 and would ensure that we are atleast on c++11?

Reply:
I tried to say that `cxx_auto_type` and `cxx_range_for` are making no difference for CMake to select C++11 (at least).

Reply:
Ok, I'll  comment on the #1428, that'll be more clear. We can close this PR, the point you raised about changing behaviour based on the cmake version is enough to not go down this road.

Reply:
Before we used the CMake `target_compile_features`, we just set the compiler flag to use C++11. Would this be a way to go forward here if the CMake version is too old? Then it is either up to CMake to choose the right flags or it is the behavior we had until recently 3.x.x?

Reply:
Maybe, I think that was also what @pboettch  suggested in his [comment](https://github.com/nlohmann/json/pull/1428#issuecomment-454297203) on #1428, the discussion here sort of got side tracked because I thought that `cxx_range_for` wasn't supported by older cmake versions. I've commented on #1428, lets try to find a resolution in that discussion.